### ISSUE TITLE
*ByDisplayValue doesn't work for Switch

### ISSUE THREAD

**Issue by jeffreyffs** [testing-library/native-testing-library#56]
- `react-native` or `expo`: 'react-native'
- `native-testing-library` version: 4.0.9
- `jest-preset`: @testing-library/react-native
- `react-native` version: 60.5
- `node` version: 10.15.1

### Relevant code or config:

```js
const Switch = ({ value, onValueChange }: {value: string; onValueChange(): void) => {
 return (<Switch value={value} onValueChange={onValueChange} />);
}

it('should work as expected', () => {
  const callback = jest.fn();
  const { getByDisplayValue } = render(<Switch value={false} onValueChanged={callback} />)

 expect(getByDisplayValue('false')).toBeTruthy()
})
```

### What you did:

Tried to write a test for a component with a Switch in it.

### What happened:

getByDisplayValue is expecting a Matcher(string | RegExp), so you can't pass in `false`, 'false' as a string also doesn't seem to work.

I also tried to ignore the Typescript error and pass `false` (and 'off') to getByDisplayValue with no luck.

Component output from `debug`
```
<Text
  style={
    Object {
      "paddingBottom": 4,
      "paddingEnd": 12,
      "paddingTop": 4,
      }
    }
>
  testLabel
</Text>
<Switch
  testID="id"
  value={false}
/>
```
### Problem description:

The [documentation](https://www.native-testing-library.com/docs/api-queries#bydisplayvalue) indicates that ByDisplayValue is what we should be using for switches.

Am I missing something?

### Can you help us fix this issue by submitting a pull request?

Possibly, if I can get some direction as to what the intended behaviour is supposed to be!

**Comment by bcarroll22** [comment]
Thanks for pointing this out, sorry for the delay. Should be fixed in c7348ec8c1a593a961076d62a99b21942948c95f

**Comment by bcarroll22** [comment]
I should point out, this fixes the functionality but not the types. I'd appreciate a PR to fix the types to accommodate the changes in that commit 👍

**Comment by bcarroll22** [comment]
FYI this is included in 4.0.10

**Comment by jeffreyffs** [comment]
I'm not sure this is working as expected. If I now pass in `false` to any of the *ByDisplayValue queries I get a crash that `matcher.test is not a function` (trying to call `test` on a boolean).

It comes from here I think:
```
      return matcher(node.getProp('value'), node, value, matchNormalizer);
```

The getProp returns a boolean for switches but I can't pass in a boolean for `value`.

**Comment by bcarroll22** [comment]
Hmm I'm not sure what's going on, when using the latest published version (4.0.10), this test works for me:

```
import React from 'react';
import { Switch } from 'react-native';
import { render } from '@testing-library/react-native';

test('it finds <Switch /> by value={false}', () => {
  const { getByDisplayValue } = render(<Switch value={false} />);

  getByDisplayValue(false);
});
```

**Comment by jeffreyffs** [comment]
🤔 I'll try a stripped down repro too

**Comment by jeffreyffs** [comment]
Hm, my stripped down repro works fine. And looking at the code... I realize it should be working. I'll figure out what is going on in my app. Thanks!

**Comment by jeffreyffs** [comment]
@bcarroll22 I've got a simple repro that breaks. It seems having a Picker in the render tree breaks things?

```
it('should render test', () => {
  const component = (
    <View>
      <Picker selectedValue="fr">
        <Picker.Item value="fr" label="French" />
      </Picker>
      <Switch value={true} />
    </View>
  );
  const {getByDisplayValue} = render(component);

  expect(getByDisplayValue(true)).toBeTruthy();```

**Comment by bcarroll22** [comment]
Okay this test passes now in v4.0.11. Hopefully that takes care of it for you 👍

**Comment by jeffreyffs** [comment]
Confirmed. Thanks so much 🎊

**Comment by jeffreyffs** [comment]
I'm back again 😢 

Unfortunately I found another combination of elements that breaks this:
```  const component = (
    <View>
      <Switch value={true} />
      <TextInput value={'44'} />
    </View>
  );
  const {getByDisplayValue} = render(component);

  expect(getByDisplayValue('44')).toBeTruthy();```

`TypeError: normalizedText.trim is not a function`

Switch and TextInput by themselves work fine.

**Comment by bcarroll22** [comment]
Hello again. Unfortunately I won't be able to look into this today. If you want to try to help debug it that would be awesome! This issue would be originating from around here: https://github.com/testing-library/native-testing-library/blob/master/src/lib/queries/display-value.js

Hope that helps if you dig into it. Thanks!

**Comment by jeffreyffs** [comment]
I'm not sure what the best approach here is. The problem is that the `matcher` is coming in as the TextInput (with a value of '44') and it is trying to compare against Switch (which has a `textToMatch` of `true`).

So it passes the `matcher` test, because it is a string and then it tries to normalize the switch value. This is where it crashes because obviously `true.trim()` won't work great.

**Comment by jeffreyffs** [comment]
All tests pass if I add a type check to `textToMatch` in `matcher.js`:
```  if (typeof matcher === 'boolean' || typeof textToMatch === 'boolean') {
    return textToMatch === matcher;
  }```

But I kind of feel like we are just applying a bandaid to a bigger problem (values might not be strings).

**Comment by jeffreyffs** [comment]
https://github.com/testing-library/native-testing-library/pull/62 for review

**Comment by bcarroll22** [comment]
Closed by #62 

Thanks!
